### PR TITLE
Damage texture modifier

### LINKED ISSUE #9607: Allow damage texture modifiers other than ^[brighten
## Problem

Sometimes you don't want the entity to flash white when you punch it. e.g. When you punch a mob you may want it to flash red instead. I propose allowing this to happen without hacky lua workarounds.

## Solutions

I propose to simply allow the entities to specify what texture modifier to use within their definition table. With a key similar to damage_texture_overlay  and "^[brighten" as the default value you would have the same functionality as before by default but if you wanted to you could set the value to be something like "^[invert:rgb" to get some really weird punches.

### PR DESCRIPTION
**Pull request by appgurueu**
- Implements damage texture modifiers
- Adds a new object property `damage_texture_modifier`
- Closes #9607

## To do

This PR is Ready for Review.

## How to test

```lua
minetest.register_on_joinplayer(function(player)
    player:set_properties{damage_texture_modifier = "^[invert:rgb"}
end)
```

### CODE REVIEW

File: src/client/content_cao.cpp [hunk 1]
```diff
@@ -1622,7 +1622,7 @@ void GenericCAO::processMessage(const std::string &data)
 				m_reset_textures_timer = 0.05;
```
Comment (sfan5) [inline src/client/content_cao.cpp:1622]: the 2 comment lines above here should be removed too

File: src/object_properties.cpp [hunk 2]
```diff
@@ -166,4 +167,5 @@ void ObjectProperties::deSerialize(std::istream &is)
 	eye_height = readF32(is);
 	zoom_fov = readF32(is);
 	use_texture_alpha = readU8(is);
+	damage_texture_modifier = deSerializeString(is);
```
Comment (sfan5) [inline src/object_properties.cpp:170]: needs backwards compatibility

File: src/client/content_cao.cpp (hunk 1 above)
Comment (appgurueu) [reply src/client/content_cao.cpp:1622]: just removed it

File: src/object_properties.cpp (hunk 2 above)
Comment (appgurueu) [reply src/object_properties.cpp:170]: sfan5: how should I implement that? checking for `is_open` ?

File: src/client/content_cao.cpp (hunk 1 above)
Comment (sfan5) [reply src/client/content_cao.cpp:1622]: Still there

Comment (appgurueu) [reply src/client/content_cao.cpp:1620]: oops, removed it in the other file but not there - now done

File: src/client/content_cao.cpp [hunk 3]
```diff
@@ -1617,12 +1617,10 @@ void GenericCAO::processMessage(const std::string &data)
 						v2f(m_prop.visual_size.X, m_prop.visual_size.Y) * BS);
 				m_env->addSimpleObject(simple);
 			} else if (m_reset_textures_timer < 0) {
```
Comment (sfan5) [inline src/client/content_cao.cpp:1619]: same for the other
```suggestion
			} else if (m_reset_textures_timer < 0 && !m_prop.damage_texture_modifier.empty()) {
```

File: doc/lua_api.txt [hunk 4]
```diff
@@ -6544,6 +6544,9 @@ Player properties need to be saved manually.
         -- deleted when the block gets unloaded.
         -- The get_staticdata() callback is never called then.
         -- Defaults to 'true'.
+
+        damage_texture_modifier = "^[brighten",
+        -- By default "^[brighten", texture modifier to be applied when object is hit
```
Comment (sfan5) [inline doc/lua_api.txt:6549]:
```suggestion
        -- Texture modifier to be applied when object is hit
```

File: src/client/content_cao.cpp (hunk 3 above)
Comment (ClobberXD) [reply src/client/content_cao.cpp:1619]: Is this to account for mods explicitly overriding the default texture modifier with `""`?

Comment (sfan5) [reply src/client/content_cao.cpp:1619]: Yes, it works fine without this but there is no need to set the texture timers and whatnot when the mod wants the damage texture disabled.

Comment (ClobberXD) [reply src/client/content_cao.cpp:1619]: Ahh, understood. Thanks.

File: src/object_properties.cpp [hunk 5]
```diff
@@ -166,4 +168,7 @@ void ObjectProperties::deSerialize(std::istream &is)
 	eye_height = readF32(is);
 	zoom_fov = readF32(is);
 	use_texture_alpha = readU8(is);
+	try {
+		damage_texture_modifier = deSerializeString(is);
+	} catch (SerializationError &e) {}
```
Comment (SmallJoker) [inline src/object_properties.cpp:173]:
```suggestion
	} catch (SerializationError &e) {
	} catch (PacketError &e) {}
```

**Pull request merged**